serial_driver/src/serial_bridge_node.cpp

```diff
@@ -0,0 +1,218 @@
+// Copyright 2021 LeoDrive, Copyright 2021 the Autoware Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "serial_driver/serial_bridge_node.hpp"
+
+#include <memory>
+#include <string>
+
+namespace lc = rclcpp_lifecycle;
+using LNI = rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface;
+using lifecycle_msgs::msg::State;
+
+namespace drivers
+{
+namespace serial_driver
+{
+
+SerialBridgeNode::SerialBridgeNode(const rclcpp::NodeOptions & options)
+: lc::LifecycleNode("serial_bridge_node", options),
+  m_owned_ctx{new IoContext(2)},
```

Comment:
does this mean there will be 2 threads always? should this be configurable? asking because honestly I dont know 😄

Reply:
This means that, if the user doesn't provide their own `IoContext` object with the number of threads pre-defined, the node will create one of its own with 2 threads. It's configurable through the `IoContext` object that the user provides if 2 isn't desired.